### DISCUSSION TITLE
.tmp removing files step giving permission denied errors when updating ATM10 to 2.8 and up

### DISCUSSION THREAD

**Discussion by fab0912** [itzg/docker-minecraft-server#3204] [Q&A] [answered]
Hello,

I've been running into an issue trying to update from ATM10 2.7 to 2.8 and higher. I'm not using the ATM10 compose file from the repo but rather the "generic pack" version with Java 21 and just tell it where the ServerFiles for ATM10 are (had issues with the Autoforge setup). It has been working totally for a few versions but now its been having issues as you can see in the image below. Its the exact same setup as with 2.7, but for some reason 2.8 and 2.9 keep throwing permission denied errors. I already tried a full clean install of it, redownloaded the zip a few times, changed permissions of the downloaded file to the docker user (also changed the whole ./data folder to the docker user, but so far nothing has worked. Just for tests I also tried "sudo docker compose up" which also throws that permission denied error.
(the error list is much longer but I decided not to include it since its all permission denied for all files in that .tmp folder)

![1 - Untitled](https://github.com/user-attachments/assets/5cec93fa-eda2-41e2-aeac-7a7c7ca3a140)

And here is the compose file:
(I removed the names from the Whitelist and OPS for privacy.)
![1 - Untitled(1)](https://github.com/user-attachments/assets/c0c61e8f-7790-41e0-a3d6-264a8692bd08)

As mentioned already, 2.7 is running perfectly fine with this setup. I'm unsure if this is some sort of bug or if my setup has a huge flaw in it and I just don't know about it.
Help would be very much appreciated and happy holidays! :D

**Comment by itzg** [comment]
It says "permission denied". Confirm and fix file ownership of the directories and files to match running server user ID, which is 1000 by default. Initial logs will also report that.

**Reply by fab0912** [reply]
I already tried that and just did it again to make sure it wasn't a fail on my end. I changed the whole folder setup ownership to the user ID 1000 and I still get the permission denied error. I'm not sure why this happens. If I do a fresh setup with ATM10 2.7 it works fine without permissions errors, but the moment I use 2.8 or 2.9 it throws those errors.

**Reply by itzg** [reply]
Not sure what more to suggest -- a permission denied error is pretty much saying it all. Make sure to use `ls -n` to see the numerical owner/group on the files/directories and add `-d` to see a directory itself. Manually remove the `./data/.tmp` (note the leading dot on `.tmp`). You could use `chown -R 1000:1000 ./data` to fix up everything in your host directory.

**Reply by fab0912** [reply]
I just copied the whole setup into a new folder and did `chown -R 1000:1000`` ./data` on it but that did nothing either. Same with removing the `.tmp`, it will copy the stuff from the zip into the .tmp again and just says permission denied.

I also a friend of mine tried it on his machine and its throwing the exact same error. So I strongly assume its not my setup at this point. 

ATM10 2.7 fresh install when it removes the files:
![2 7_working](https://github.com/user-attachments/assets/6d4b2043-8a0b-418c-a51d-7efd7ddcf7dd)

ATM10 2.9 fresh install when it removes the files:
![2 9_failed](https://github.com/user-attachments/assets/fdb77462-6ada-4517-8bb5-8c666a620cff)

**Reply by itzg** [reply]
Oh, ATM10? Please just use the example setup that several people are using 

https://github.com/itzg/docker-minecraft-server/blob/master/examples/atm10/docker-compose.yml

**Reply by fab0912** [reply, marked as answer]
I know, but like I mentioned before, my CurseForge API key keeps getting limited and the downloads stop, so that AutoForge isn't working for me.
Thats why I'm using the generic setup. But anyway, we might have found the solution to what was going on. Basically the server-files.zip of ATM 10 has very weird permissions inside (version 2.8 and 2.9). Basically the whole content of the zip in 2.7 is writable and 2.8+2.9 had files that were read only.

So it would be awesome if you could add something like this: when you unzip the server files in your script you `chmod 755` them so any weirdness with permissions being carried over can be prevented, at least if that doesn't cause any security implications.

Otherwise the user has to manually unzip -> change permission -> zip -> copy into the downloads folder again

**Reply by itzg** [reply]
Great find with the zip contents and great suggestion. That enhancement makes sense.
